axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetImportServiceImpl.java

```diff
@@ -81,18 +85,31 @@ public FixedAsset generateAndComputeLines(
     if (isImported(fixedAsset)) {
 
       if (fixedAsset.getDisposalDate() == null) {
-        BigDecimal grossValue = fixedAsset.getGrossValue();
-        BigDecimal alreadyDepreciatedAmount = fixedAsset.getImportAlreadyDepreciatedAmount();
-        BigDecimal depreciatedAmountCurrentYear = fixedAsset.getDepreciatedAmountCurrentYear();
+        BigDecimal grossValue =
+            currencyScaleServiceAccount.getCompanyScaledValue(
+                fixedAsset, fixedAsset.getGrossValue());
+        BigDecimal alreadyDepreciatedAmount =
+            currencyScaleServiceAccount.getCompanyScaledValue(
+                fixedAsset, fixedAsset.getImportAlreadyDepreciatedAmount());
+        BigDecimal depreciatedAmountCurrentYear =
+            currencyScaleServiceAccount.getCompanyScaledValue(
+                fixedAsset, fixedAsset.getDepreciatedAmountCurrentYear());
         boolean isTotallyDepreciated =
-            grossValue.equals(alreadyDepreciatedAmount.add(depreciatedAmountCurrentYear))
-                || grossValue
-                    .subtract(fixedAsset.getResidualValue())
-                    .equals(alreadyDepreciatedAmount.add(depreciatedAmountCurrentYear));
+            grossValue.equals(
+                    currencyScaleServiceAccount.getCompanyScaledValue(
+                        fixedAsset, alreadyDepreciatedAmount.add(depreciatedAmountCurrentYear)))
+                || currencyScaleServiceAccount
+                    .getCompanyScaledValue(
+                        fixedAsset, grossValue.subtract(fixedAsset.getResidualValue()))
+                    .equals(
+                        currencyScaleServiceAccount.getCompanyScaledValue(
+                            fixedAsset,
+                            alreadyDepreciatedAmount.add(depreciatedAmountCurrentYear)));
```

Comment:
I'm sure sure that `equals` should be used for BigDecimal. Anyways, you can use the new method form before

Reply:
Resolved